controllers/kyma_controller.go

```diff
@@ -152,7 +153,7 @@ func (r *KymaReconciler) deleteKyma(ctx context.Context, kyma *v1beta1.Kyma) (ct
 	// if the status is not yet set to deleting, also update the status of the control-plane
 	// in the next sync cycle
 	if err := status.Helper(r).UpdateStatusForExistingModules(
-		ctx, kyma, v1beta1.StateDeleting, "waiting for modules to be deleted",
+		ctx, kyma, v1beta1.StateDeleting, "waiting for modules to be deleted", r.IsKymaManaged,
```

Comment:
`r.IsKymaManaged` is not changing between calls to  `UpdateStatusForExistingModules`, the same value is always used - either `true` or `false`, because that is coming from the flag.
Extending the method signature with  a parameter that has always the same value for the entire life of a program is not a good design.
Because the Helper already accepts `r` - which contains information about `isKymaManaged` - perhaps this knowledge could be used in the implementation of `UpdateStatusForExistingModules`, instead of passing that value explicitly?